src/librustc_typeck/check/mod.rs

```diff
@@ -1593,7 +1593,7 @@ impl<'a, 'gcx, 'tcx> AstConv<'gcx, 'tcx> for FnCtxt<'a, 'gcx, 'tcx> {
                 infer::LateBoundRegionConversionTime::AssocTypeProjection(item_name),
                 &poly_trait_ref);
 
-        self.tcx().mk_projection(trait_ref, item_name)
+        self.tcx().mk_deprecated_projection(trait_ref, item_name)
```

Comment:
Done. Feels like I could keep pulling and perhaps remove `item_name`, too - but perhaps not this time around.